matchzoo/models/drmm.py

```diff
@@ -0,0 +1,107 @@
+"""An implementation of DRMM Model."""
+import typing
+
+import keras
+import keras.backend as K
+from keras.activations import softmax
+
+from matchzoo import engine
+
+
+class DRMM(engine.BaseModel):
+    """
+    DRMM Model.
+
+    Examples:
+        >>> model = DRMM()
+        >>> model.guess_and_fill_missing_params(verbose=0)
+        >>> model.build()
+
+    """
+
+    @classmethod
+    def get_default_params(cls) -> engine.ParamTable:
+        """:return: model default parameters."""
+        params = super().get_default_params(with_embedding=True)
+        params['optimizer'] = 'adam'
+        params['input_shapes'] = [(5,), (5, 30,)]
+        params.add(engine.Param('hidden_sizes', [5, 1]))
+        return params
+
+    def build(self):
+        """Build model structure."""
+
+        # Scalar dimensions referenced here:
+        #   B = batch size (number of sequences)
+        #   D = embedding size
+        #   L = `input_left` sequence length
+        #   R = `input_right` sequence length
+        #   K = size of top-k
+
+        # Left input and right input.
+        # shape = [B, L]
+        # shape = [B, R]
+        query, doc = self._make_inputs()
```

Comment:
Should the shape of doc be [B, dimension of histogram] ? The input should still be a pre-computed interaction matching matrix, correct ?